src/communication.rs

```diff
@@ -114,13 +181,64 @@ impl UPayload {
         }
     }
 
+    /// Creates a new UPayload from a protobuf `Message`, will set payload
+    /// type to `UPayload::UPAYLOAD_FORMAT_PROTOBUF`.
+    ///
+    /// # Arguments
+    ///
+    /// * `message` - The protobuf `Message` to wrap in UPayload.
+    ///
+    /// # Returns
+    ///
+    /// `UPayload` with `Message` as `UPayloadType::UPAYLOAD_FORMAT_PROTOBUF`` payload.
+    pub fn from_protobuf<M>(message: M) -> Option<Self>
```

Comment:
IMHO this should better return a Result with an error if the message cannot be serialized. However, we might also consider implementing `TryFrom<M: Message>` instead ...

Reply:
Can't make the TryFrom work, conflict with generic implemenation from core... went for using Result<>.